libs/frontend/util-edit-text/src/lib/base-keymap.ts

```diff
@@ -0,0 +1,995 @@
+  }
+  return true
+}
+
+/// If the cursor is in an empty textblock that can be lifted, lift the
+/// block.
+export const liftEmptyBlock: Command = (state, dispatch) => {
+  let { $cursor } = state.selection as TextSelection
+  if (!$cursor || $cursor.parent.content.size) return false
+  if ($cursor.depth > 1 && $cursor.after() != $cursor.end(-1)) {
+    let before = $cursor.before()
+    if (canSplit(state.doc, before)) {
+      if (dispatch) dispatch(state.tr.split(before).scrollIntoView())
+      return true
+    }
+  }
+  let range = $cursor.blockRange(),
+    target = range && liftTarget(range)
+  if (target == null) return false
+  if (dispatch) dispatch(state.tr.lift(range!, target).scrollIntoView())
+  return true
+}
+
+/// Create a variant of [`splitBlock`](#commands.splitBlock) that uses
+/// a custom function to determine the type of the newly split off block.
+export function splitBlockAs(
+  splitNode?: (node: Node, atEnd: boolean) => { type: NodeType; attrs?: Attrs } | null,
+): Command {
+  return (state, dispatch) => {
+    let { $from, $to } = state.selection
+    if (state.selection instanceof NodeSelection && state.selection.node.isBlock) {
+      if (!$from.parentOffset || !canSplit(state.doc, $from.pos)) return false
+      if (dispatch) dispatch(state.tr.split($from.pos).scrollIntoView())
+      return true
+    }
+
+    if (!$from.parent.isBlock) return false
+
+    if (dispatch) {
+      let atEnd = $to.parentOffset == $to.parent.content.size
+      let tr = state.tr
+      if (
+        state.selection instanceof TextSelection ||
+        state.selection instanceof AllSelection
+      )
+        tr.deleteSelection()
+      let deflt =
+        $from.depth == 0
+          ? null
+          : defaultBlockAt($from.node(-1).contentMatchAt($from.indexAfter(-1)))
+      let splitType = splitNode && splitNode($to.parent, atEnd)
+      let types = splitType ? [splitType] : atEnd && deflt ? [{ type: deflt }] : undefined
+      let can = canSplit(tr.doc, tr.mapping.map($from.pos), 1, types)
+      if (
+        !types &&
+        !can &&
+        canSplit(
+          tr.doc,
+          tr.mapping.map($from.pos),
+          1,
+          deflt ? [{ type: deflt }] : undefined,
+        )
+      ) {
+        if (deflt) types = [{ type: deflt }]
+        can = true
+      }
+      if (can) {
+        tr.split(tr.mapping.map($from.pos), 1, types)
+        if (!atEnd && !$from.parentOffset && $from.parent.type != deflt) {
+          let first = tr.mapping.map($from.before()),
+            $first = tr.doc.resolve(first)
+          if (
+            deflt &&
+            $from.node(-1).canReplaceWith($first.index(), $first.index() + 1, deflt)
+          )
+            tr.setNodeMarkup(tr.mapping.map($from.before()), deflt)
+        }
+      }
+      // Originally this was `dispatch(tr.scrollIntoView())`. `scrollIntoView()` was removed
+      // owing to https://github.com/samatechtw/pubstudio-builder/issues/173.
+      dispatch(tr)
```

Comment:
Hi @sampullman 

This file is a copied of the `baseKeymap` from `prosemirror-commands`. But I have to make an adjustment here to solve the bug found in #173.